tests/CppWinRTComServerTests.cpp

```diff
@@ -129,32 +132,49 @@ TEST_CASE("CppWinRTComServerTests::AnyRegisterFailureClearAllRegistrations", "[c
 
 TEST_CASE("CppWinRTComServerTests::NotifierAndRegistration", "[cppwinrt_com_server]")
 {
-    wil::unique_event moduleEvent(wil::EventOptions::None);
-    wil::unique_event coroutineRunning(wil::EventOptions::None);
-    wil::unique_event coroutineContinue(wil::EventOptions::None);
+    wil::unique_event moduleEvent(wil::EventOptions::ManualReset);
+    wil::unique_event coroutineRunning(wil::EventOptions::ManualReset);
+    wil::unique_event coroutineContinue(wil::EventOptions::ManualReset);
 
     wil::notifiable_module_lock::instance().set_notifier([&]() {
         moduleEvent.SetEvent();
     });
+    auto resetOnExit = wil::scope_exit([&] {
+        wil::notifiable_module_lock::instance().set_notifier(nullptr);
+    });
 
     winrt::init_apartment();
 
     auto revoker = wil::register_com_server<MyServer>();
 
-    [&]() -> winrt::Windows::Foundation::IAsyncAction {
-        co_await winrt::resume_background();
-        coroutineRunning.SetEvent();
+    std::exception_ptr coroutineException;
+    auto asyncLambda = [&]() -> winrt::Windows::Foundation::IAsyncAction {

```

Comment:
_Consider_ commenting that the lambda needs a variable so that the captures live until the wait that the coroutine is done executing.  I almost commented that "you don't need this variable".  That would be true of a non-coroutine lambda.  But it is very much not true here.  I think this subtlety is worth a small comment.